homeassistant/components/knx/fan.py

```diff
@@ -68,6 +69,11 @@ def percentage(self) -> Optional[int]:
             )
         return self._device.current_speed
 
+    @property
+    def speed_count(self) -> Optional[int]:
+        """Return the number of speeds the fan supports."""
+        return int_states_in_range(self._step_range)
```

Comment:
Does `int_states_in_range` handle `None` gracefully? `self._step_range` can be None, if the KNX fan supports setting percentage directly.

Reply:
No, it will blow up. The guard should be in the integration though and not the util

Reply:
I added a guard to return the maximum allowed steps in this case (100)